src/transformers/models/megatron_gpt2/convert_megatron_gpt2_checkpoint.py

```diff
@@ -97,17 +99,16 @@ def convert_megatron_checkpoint(args, input_state_dict):
 
     # The word embeddings.
     word_embeddings = embeddings["word_embeddings"]["weight"]
-    # Truncate the embedding table to 50257 rows.
-    word_embeddings = word_embeddings[:50257, :]
-    # Truncate the embedding table to 50257 rows.
+    # Truncate the embedding table to vocab_size rows.
+    word_embeddings = word_embeddings[: config.vocab_size, :]
     output_state_dict["transformer.wte.weight"] = word_embeddings
 
     # The position embeddings.
     pos_embeddings = embeddings["position_embeddings"]["weight"]
     # Read the hidden dimension.
-    hidden_size = pos_embeddings.size(0)
+    n_embed = pos_embeddings.size(0)
     # DEBUG.
-    assert hidden_size == heads * hidden_size_per_head
+    assert n_embed == heads * hidden_size_per_head
```

Comment:
It's just a minor comment but this assertion seems no longer necessary since we have `hidden_size_per_head = config.n_embd // config.n_head`at line 86.

Reply:
Hi!

The line asserts that the model size read from the shape of `embeddings["position_embeddings"]["weight"]` matches the size read in the config file.